### PR TITLE
Give Paramedics Emergency JoL

### PR DESCRIPTION
**Pull request by SolStar2**
## About the PR

Give paramedics Emergency Jaws of Life round start. Emergency JoL have no cutting and requires power cell to use. Medium power cell gives 5 uses. Antags get 2 less tc to compensate. Research and CE get advanced JoL which maintain original functionality.

More loadout changes for space
 - remove body bag from paramedics backpack and put a body bag box in locker
 - removed GPS as one is in locker
 - moved handheld monitor to locker
 - whitelist defibs on emt belt and moved portafib there

## Why / Balance

Jaws of Life in real life are used by first responders to get to places with critically injured people so this is a very natural part of their kit. Paramedics need more tools to be effective first responders. Currently, they are expected to retrieve critical people from places they cannot get to. This gives them the real-life solution to that already exists in game. Advanced tools are already worthwhile research to get good tools to engis nicer tools.

## Requirements

- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl: SolStar
- tweak: Paramedics now get Jaws of Life in order to get into places in an emergency

### CODE REVIEW

**Comment by deltanedas** [comment]
so much stuff in the bag now it will probably overflow a backpack or satchel

**Comment by SolStar2** [comment]
You're right I moved the body bag to the locker so they have space.

File: Resources/Prototypes/Catalog/Fills/Lockers/medical.yml [hunk 1]
```diff
@@ -185,6 +185,7 @@
       - id: ClothingHeadsetMedical
       - id: ClothingMaskSterile
       - id: ClothingShoesBootsWinterParamedic #Delta V: Add departmental winter boots
+      - id: BodyBagFolded #Delta V: Give paramedics a body bag
```
Comment (deltanedas) [inline Resources/Prototypes/Catalog/Fills/Lockers/medical.yml]: `# DeltaV -`

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by Avalon-Proto** [comment]
> You're right I moved the body bag to the locker so they have space.

Honestly you might wanna do that w/ the portafib too, or let their belt hold the jaws. Cause if they have any trinkets it's gonna be more space issues

**Comment by Avalon-Proto** [comment]
Would be kinda neat to have a unique sprite for the Paramed jaws of life tho

**Comment by SolStar2** [comment]
> > You're right I moved the body bag to the locker so they have space.
> 
> Honestly you might wanna do that w/ the portafib too, or let their belt hold the jaws. Cause if they have any trinkets it's gonna be more space issues

PortaFib may make sense on the belt

**Comment by SolStar2** [comment]
> > You're right I moved the body bag to the locker so they have space.
> 
> Honestly you might wanna do that w/ the portafib too, or let their belt hold the jaws. Cause if they have any trinkets it's gonna be more space issues

Also this was tested with maximum space using trinkets but more may be added so that is a fair point

**Comment by Avalon-Proto** [comment]
> Also this was tested with maximum space using trinkets but more may be added so that is a fair point

The paramedic who spawns with a case of cigars: 

But yeah honestly both corpsman and paramedic have a huge amount of stuff in their bags that should really be split into the lockers too or something :p

**Comment by SolStar2** [comment]
> Would be kinda neat to have a unique sprite for the Paramed jaws of life tho

This would be nice or even make the research one a more powerful and advanced version but wanted to test the waters here.

File: Resources/Prototypes/Catalog/Fills/Lockers/medical.yml [hunk 2]
```diff
@@ -208,6 +209,7 @@
       - id: ClothingHeadsetMedical
       - id: ClothingMaskSterile
       - id: ClothingShoesBootsWinterParamedic #Delta V: Add departmental winter boots
+      - id: BoxBodyBag # Delta V - Give paramedics a body bag
```
Comment (deltanedas) [inline Resources/Prototypes/Catalog/Fills/Lockers/medical.yml]: DeltaV

File: Resources/Prototypes/Catalog/Fills/Lockers/medical.yml [hunk 3]
```diff
@@ -185,13 +185,14 @@
...
       - id: MedkitFilled
         prob: 0.3
       - id: LunchboxMedicalFilledRandom # Delta-V Lunchboxes!
         prob: 0.3
 
 - type: entity
-  id: LockerParamedicFilledHardsuit #Delta V - Paramedic locker with suit
+  id: LockerParamedicFilledHardsuit # Delta V - Paramedic locker with suit
```
Comment (deltanedas) [inline Resources/Prototypes/Catalog/Fills/Lockers/medical.yml]: DeltaV

File: Resources/Prototypes/Catalog/Fills/Lockers/medical.yml [hunk 4]
```diff
@@ -185,13 +185,14 @@
       - id: ClothingHeadsetMedical
       - id: ClothingMaskSterile
       - id: ClothingShoesBootsWinterParamedic #Delta V: Add departmental winter boots
+      - id: BoxBodyBag # Delta V - Give paramedics a body bag
```
Comment (deltanedas) [inline Resources/Prototypes/Catalog/Fills/Lockers/medical.yml]: DeltaV

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by Quanteey** [comment]
My two cents would be to make the JoL parameds get broadcast its location when it is being used in medical comms. That way it is more controlled, the CMO can oversee its usage, and prevents abuse in situations where the paramedic might not be an antagonist.

**Review by deltanedas — DISMISSED** [review]

**Comment by SolStar2** [comment]
Made changes per directions request, still undergoing review.

**Comment by deltanedas** [comment]
i dont see the point of these changes:
- removing wirecutter is utterly pointless, wirecutters are plentiful and have nothing to do with breaking into places
- batteries are a dime a dozen you can find them EVERYWHERE so this would just become annoying like a holofan running out of charge and you dont stockpile batteries from maints

also in general AI exists which parameds can nag, or they can ask engi to print actual jaws which are usually unlocked 20-30m in

**Comment by RupertMonroe1** [comment]
Jaws of life are incredibly powerful tools and I don't really see a good reason for paramedics to starts with them, you have so many other ways to get into rooms that these aren't really needed. Just wait until EPI researches advanced tools and ask for a permit

**Comment by chamomileteatime** [comment]
direction approved!

**Comment by chamomileteatime** [comment]
Per deltanedas’ concerns:
IMG_1694
IMG_1695

**Review by chamomileteatime — APPROVED** [review]

**Comment by Avalon-Proto** [comment]
I got a question, are these still 2x2 or since they are more limited, are they possibly 1x2?

**Comment by SolStar2** [comment]
> I got a question, are these still 2x2 or since they are more limited, are they possibly 1x2?

I left it as a 2x2 since it being a 2x2 is already a stretch honestly that is mentioned in the original JoL description.

File: Content.Shared/Prying/Systems/PryingSystem.cs [hunk 5]
```diff
@@ -71,6 +74,10 @@ public bool TryPry(EntityUid target, EntityUid user, out DoAfterId? id, EntityUi
         if (!comp.Enabled)
             return false;
 
+        // DeltaV - Cancel if no power and needs power
+        if (TryComp<PryingUsePowerComponent>(tool, out var powerComp) && !_power.HasCharge(tool, powerComp.UseCost, null, user))
+                return false;
```
Comment (deltanedas) [inline Content.Shared/Prying/Systems/PryingSystem.cs]: remove, raise the event in CanPry on the tool as well so this component can handle it

**Review by deltanedas — CHANGES_REQUESTED** [review]
power changes are a conflict minefield, use HasActivatableCharge instead with the power use moved to power cell draw component on jaws

**Comment by SolStar2** [comment]
> power changes are a conflict minefield, use HasActivatableCharge instead with the power use moved to power cell draw component on jaws

I can revert the changes to power system however HasActivatiableCharge makes some assumptions that cause its power use on activation work mostly with UIs. I am referring to how items like stun batons handle it's usage of charge it puts the power use into its own component for it to be handled by its system.

**Comment by SolStar2** [comment]
Only issue at the moment is there is no prediction for the client but not quite sure how to fix that.

File: Content.Shared/Prying/Systems/PryingSystem.cs [hunk 6]
```diff
@@ -71,7 +71,7 @@ public bool TryPry(EntityUid target, EntityUid user, out DoAfterId? id, EntityUi
         if (!comp.Enabled)
             return false;
 
-        if (!CanPry(target, user, out var message, comp))
+        if (!CanPry(target, user, out var message, comp, null, tool)) // DeltaV - Add tool
```
Comment (deltanedas) [inline Content.Shared/Prying/Systems/PryingSystem.cs]: use tool: tool instead of a null before it

File: Content.Shared/Prying/Systems/PryingSystem.cs [hunk 7]
```diff
@@ -124,6 +128,11 @@ private bool CanPry(EntityUid target, EntityUid user, out string? message, Pryin
 
         RaiseLocalEvent(target, ref canev);
 
+        // Begin DeltaV - Raise event on tool if one is being used.
+        if (tool != null)
+            RaiseLocalEvent((EntityUid)tool, ref canev);
```
Comment (deltanedas) [inline Content.Shared/Prying/Systems/PryingSystem.cs]: tool is {} toolUid instead of evil cast

File: Content.Shared/Prying/Systems/PryingSystem.cs [hunk 8]
```diff
@@ -167,7 +176,7 @@ private void OnDoAfter(EntityUid uid, DoorComponent door, DoorPryDoAfterEvent ar
 
         TryComp<PryingComponent>(args.Used, out var comp);
 
-        if (!CanPry(uid, args.User, out var message, comp))
+        if (!CanPry(uid, args.User, out var message, comp, null, args.Used)) // DeltaV - Check tool for power
```
Comment (deltanedas) [inline Content.Shared/Prying/Systems/PryingSystem.cs]: same use tool: args.Used

File: Content.Shared/Prying/Systems/PryingSystem.cs [hunk 9]
```diff
@@ -181,6 +190,10 @@ private void OnDoAfter(EntityUid uid, DoorComponent door, DoorPryDoAfterEvent ar
 
         var ev = new PriedEvent(args.User);
         RaiseLocalEvent(uid, ref ev);
+        // Begin DeltaV - Emergency JoL uses power.
+        if (args.Used != null)
+            RaiseLocalEvent((EntityUid)args.Used, ref ev);
```
Comment (deltanedas) [inline Content.Shared/Prying/Systems/PryingSystem.cs]: is {} tool

File: Resources/Prototypes/Entities/Objects/Tools/jaws_of_life.yml
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Tools/jaws_of_life.yml:1]: instead of this just have emergency jol redefine it to only having prying in MultipleTool and adding the power usage

File: Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+- type: entity
+  name: emergency jaws of life
+  parent: [BaseJawsOfLife, PowerCellSlotMediumItem]
+  id: EmergencyJawsOfLife
+  description: A set of Jaws of Life. EMTs may use them to pry open powered airlocks in an emergency.
+  components:
+  - type: Sprite
+    sprite: _DV/Objects/Tools/emergency_jaws_of_life.rsi
+    state: jaws_pry
```
Comment (deltanedas) [inline Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml]: inherited

File: Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  - type: Clothing
+    sprite: _DV/Objects/Tools/emergency_jaws_of_life.rsi
+    quickEquip: false
+    slots:
+    - Belt
+  - type: ToolTileCompatible
```
Comment (deltanedas) [inline Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml]: inherited

File: Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+  - type: Prying
+    speedModifier: .75 # High torque low speed, Half speed of proper JoL
```
Comment (deltanedas) [inline Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml]: thus is insanely slow for powered doors, any half decent ai would be with you much faster

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by SolStar2** [comment]
1x speed should probably be fine

File: Content.Server/_DV/Tools/PryingRequiresPowerComponent.cs [hunk 13]
```diff
@@ -0,0 +1,8 @@
+namespace Content.Server._DV.Tools;
+
+[RegisterComponent]
+public sealed partial class PryingRequiresPowerComponent : Component
+{
+    [DataField]
+    public float PowerCost = 0;
```
Comment (deltanedas) [inline Content.Server/_DV/Tools/PryingRequiresPowerComponent.cs]: 0 is the default dont need it

File: Content.Server/_DV/Tools/PryingRequiresPowerComponent.cs [hunk 14]
```diff
@@ -0,0 +1,8 @@
+namespace Content.Server._DV.Tools;
+
+[RegisterComponent]
```
Comment (deltanedas) [inline Content.Server/_DV/Tools/PryingRequiresPowerComponent.cs]: add `Access(typeof(PryingRequiresPowerSystem))`

File: Content.Server/_DV/Tools/PryingRequiresPowerSystem.cs [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+using Content.Server.PowerCell;
+using Content.Shared.Prying.Components;
+
+namespace Content.Server._DV.Tools;
+
+public sealed class PryingRequiresPowerSystem : EntitySystem
+{
+    [Dependency] private readonly PowerCellSystem _cell = default!;
+    [Dependency] private readonly BatterySystem _battery = default!;
+    public override void Initialize()
```
Comment (deltanedas) [inline Content.Server/_DV/Tools/PryingRequiresPowerSystem.cs:13]: add newline between dependencies and initialize

File: Content.Server/_DV/Tools/PryingRequiresPowerSystem.cs [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+
+    private void OnPried(EntityUid uid, PryingRequiresPowerComponent component, PriedEvent args)
+    {
+        BatteryComponent? batteryComp;
+        // Tool is a battery use its power
+        if (TryComp<BatteryComponent>(uid, out batteryComp))
+        {
+            _battery.UseCharge(uid, component.PowerCost, batteryComp);
+            return;
+        }
```
Comment (deltanedas) [inline Content.Server/_DV/Tools/PryingRequiresPowerSystem.cs]: use TryUseCharge instead of meme trycomp that wasnt using out var

File: Content.Server/_DV/Tools/PryingRequiresPowerSystem.cs [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+    [Dependency] private readonly BatterySystem _battery = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<PryingRequiresPowerComponent, BeforePryEvent>(OnBeforePry);
+        SubscribeLocalEvent<PryingRequiresPowerComponent, PriedEvent>(OnPried);
+    }
+
+    private void OnPried(EntityUid uid, PryingRequiresPowerComponent component, PriedEvent args)
```
Comment (deltanedas) [inline Content.Server/_DV/Tools/PryingRequiresPowerSystem.cs]: change subscriptions to `Entity<PryingRequiresPowerComponent> ent, ref T args`

File: Resources/Prototypes/Entities/Objects/Tools/jaws_of_life.yml [hunk 18]
```diff
@@ -1,8 +1,8 @@
 - type: entity
-  name: jaws of life
+  name: advanced jaws of life # DeltaV - Renamed to advanced jaws of life, basic jaws of life now requires power
   parent: BaseItem
   id: JawsOfLife
-  description: A set of jaws of life, compressed through the magic of science.
+  description: An improved set of jaws of life, now with a built in microreactor and a configurable head for wire cutting. # DeltaV - Renamed to advanced jaws of life, basic jaws of life now requires power
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Tools/jaws_of_life.yml]: do this in ftl instead

File: Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+  - type: Sprite
+    sprite: _DV/Objects/Tools/emergency_jaws_of_life.rsi
+  - type: Clothing
+    sprite: _DV/Objects/Tools/emergency_jaws_of_life.rsi
+  - type: ToolTileCompatible
+    delay: .75
+  - type: Prying
+    speedModifier: 1 # High torque low speed, Half speed of proper JoL
+  - type: MultipleTool
+    statusShowBehavior: true
```
Comment (deltanedas) [inline Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml]: this is inherited just have entries

File: Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml [hunk 20]
```diff
@@ -0,0 +1,24 @@
+- type: entity
+  name: emergency jaws of life
```
Comment (deltanedas) [inline Resources/Prototypes/_DV/Entities/Objects/Tools/jaws_of_life.yml]: move name below id

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/_DV/tools/jaws-of-life.ftl
Comment (deltanedas) [inline Resources/Locale/en-US/_DV/tools/jaws-of-life.ftl:1]: move to a subdirectory prototypes/entity/tools

**Comment by SolStar2** [comment]
Its ready for review again :3

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Review by deltanedas — APPROVED** [review]
